odl/tomo/backends/astra_setup.py

```diff
@@ -202,79 +299,39 @@ def astra_geom_to_vec(geometry):
-
-    elif isinstance(geometry, Parallel3dGeometry):
-        vectors = np.zeros((num_angles, 12))
-        det_pix_width = geometry.det_grid.stride[0]
-        det_pix_height = geometry.det_grid.stride[1]
-
-        for ang_idx, angle in enumerate(angles.points()):
-            # ray direction
-            vectors[ang_idx, 0] = sin(angle)
-            vectors[ang_idx, 1] = -cos(angle)
-            # vectors[nn, 2] = 0
-
-            # center of detector
-            # vectors[nn, 3:6] = 0
-
-            # vector from detector pixel (0,0) to (0,1)
-            # TODO: use det_rotation method instead of
-            vectors[ang_idx, 6] = cos(angle) * det_pix_width
-            vectors[ang_idx, 7] = sin(angle) * det_pix_width
-            # vectors[nn, 8] = 0
-
-            # vector from detector pixel (0,0) to (1,0)
-            # vectors[nn, 9] = 0
-            # vectors[nn, 10] = 0
-            vectors[ang_idx, 11] = det_pix_height
-
-    elif isinstance(geometry, FanBeamGeometry):
-        vectors = np.zeros((num_angles, 6))
-        det_pix_width = geometry.det_grid.stride[0]
-
-        for ang_idx, angle in enumerate(angles.points()):
-            # source position
-            # TODO: check method, probably inconsistent with detector pixel vec
-            vectors[ang_idx, 0:2] = geometry.src_position(angle)
-
-            # center of detector
-            # TODO: check method, probably inconsistent with detector pixel vec
-            vectors[ang_idx, 2:4] = geometry.det_refpoint(angle)
-
-            # vector from detector pixel (0,0) to (0,1)
-            # TODO: use `det_rotation` method instead of
-            vectors[ang_idx, 4] = cos(angle) * det_pix_width
-            vectors[ang_idx, 5] = sin(angle) * det_pix_width
+    angles = geometry.motion_grid
+    vectors = np.zeros((angles.ntotal, 12))
 
-    else:
-        raise ValueError('invalid geometry type {!r}.'.format(geometry))
+    for ang_idx, angle in enumerate(angles.points()):
+        rot_matrix = geometry.rotation_matrix(angle)
+
+        midp = geometry.det_params.midpoint
+
+        # source position
+        vectors[ang_idx, 0:3] = geometry.det_to_src(angle, midp)
+
+        # center of detector
+        vectors[ang_idx, 3:6] = geometry.det_point_position(angle, midp)
+
+        # vector from detector pixel (0,0) to (0,1)
+        unit_vecs = geometry.detector.surface_deriv()
+        strides = geometry.det_grid.stride
+        vectors[ang_idx, 6:9] = rot_matrix.dot(unit_vecs[0] * strides[0])
+        vectors[ang_idx, 9:12] = rot_matrix.dot(unit_vecs[1] * strides[1])
+
+    # Astra order, needed for data to match what we expect from astra.
+    # Astra has a different axis convention to ODL (z, y, x), so we need
+    # to adapt to this by changing the order
+
+    newind = []
+    for i in range(4):
+        newind += [2 + 3 * i, 1 + 3 * i, 0 + 3 * i]
+    vectors = vectors[:, newind]
+
+    if np.any(vectors[:, 0:3] == 0):
```

Comment:
I just tested the with the latest ASTRA version and it works. So can be removed.